packages/db/src/core/cli/commands/link/index.ts

```diff
@@ -16,20 +16,16 @@ export async function cmd() {
 		console.error(MISSING_SESSION_ID_ERROR);
 		process.exit(1);
 	}
-	const getWorkspaceIdAsync = getWorkspaceId().catch((err) => {
-		return err as Error;
-	});
+	const workspaceId = await promptWorkspace(sessionToken);
```

Comment:
This is prompting too early in the flow, if you run this code you'll see it prompt for the workspace before it prompts to ask if you even want to link the project in the first place (`promptBegin()`)

I imagine, as a user, you'd want:

- `promptBegin`
- `promptLinkExisting`
- `promptWorkspace`
- then, fork between `isLinkExisting`, `isLinkNew`,  etc.

Basically, just move this down to like 22 and you should be good to go